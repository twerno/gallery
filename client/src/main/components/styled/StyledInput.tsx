import styled from 'styled-components';

export default styled.input`
    padding: 9px 18px;
    width: 100%;
    border: .75px solid #cecece;
    font-size: 1rem;

    &:focus {
        outline-color: rgb(103, 156, 226);
    }

`;
