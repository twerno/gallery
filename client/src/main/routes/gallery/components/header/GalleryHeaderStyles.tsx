import styled from 'styled-components';

const SearchForm = styled.form`
    width: 100%;
    display: flex;
    flex-direction: row;
    padding: 10px 10px;
    background-color: white;
`;


export default {
    SearchForm
}