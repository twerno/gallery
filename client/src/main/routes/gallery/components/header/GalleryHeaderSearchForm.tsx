import { useFormik } from 'formik';
import StyledButton from 'main/components/styled/StyledButton';
import StyledInput from 'main/components/styled/StyledInput';
import * as React from 'react';
import { IGalleryUrlQuery } from '../../model/galleryQuery';
import GalleryHeaderStyles from './GalleryHeaderStyles';

export interface IGalleryHeaderSearchFormProps {
    initialValues: IGalleryUrlQuery;
    onSubmit: (query: IGalleryUrlQuery) => void;
}

export const GalleryHeaderSearchForm = ({ initialValues, onSubmit }: IGalleryHeaderSearchFormProps) => {

    const { handleChange, handleSubmit, values, isSubmitting } = useFormik({
        initialValues,
        onSubmit
    });

    return (
        <GalleryHeaderStyles.SearchForm onSubmit={handleSubmit}>

            <StyledInput
                id="q"
                name="q"
                aria-label="Search"
                type="text"
                onChange={handleChange}
                value={values.q || ''}
            />

            <StyledButton
                mode='inputButton'
                type="submit"
                disabled={isSubmitting}
            >
                Search
            </StyledButton>

        </GalleryHeaderStyles.SearchForm>
    );
}