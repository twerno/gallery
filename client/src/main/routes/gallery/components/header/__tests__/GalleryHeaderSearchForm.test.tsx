import '@testing-library/jest-dom/extend-expect';
import { wait } from '@testing-library/react';
import { shallow, mount } from 'enzyme';
import 'jest-styled-components';
import StyledInput from 'main/components/styled/StyledInput';
import * as React from 'react';
import { componentBasicTests } from 'test/componentBasicTests';
import { GalleryHeaderSearchForm, IGalleryHeaderSearchFormProps } from '../GalleryHeaderSearchForm';
import GalleryHeaderStyles from '../GalleryHeaderStyles';

describe('GalleryHeaderSearchForm', () => {

    const props: IGalleryHeaderSearchFormProps = {
        initialValues: { q: 'cats' },
        onSubmit: (_) => { },
    };

    componentBasicTests(
        "GalleryHeaderSearchForm",
        <GalleryHeaderSearchForm {...props} />
    );

    componentBasicTests(
        "GalleryHeaderSearchForm",
        <GalleryHeaderSearchForm {...props} initialValues={{ q: undefined }} />
    );

    test("submitting", async () => {
        const onSubmit = jest.fn();
        const wraper = shallow(<GalleryHeaderSearchForm initialValues={{ q: 'cats' }} onSubmit={onSubmit} />);

        await wait(() => wraper.find(GalleryHeaderStyles.SearchForm).simulate('submit', {}));
        expect(onSubmit).toBeCalledTimes(1);
        expect(onSubmit).toBeCalledWith({ q: 'cats' }, expect.anything());
    });

    test("edit query and submit", async () => {
        const onSubmit = jest.fn();
        const wraper = mount(<GalleryHeaderSearchForm initialValues={{ q: undefined }} onSubmit={onSubmit} />);

        const input = wraper.find(StyledInput);
        await wait(() => input.simulate('change', { target: { name: "q", value: 'cats' } }));

        const form = wraper.find(GalleryHeaderStyles.SearchForm);
        await wait(() => form.simulate('submit', {}));
        expect(onSubmit).toBeCalledTimes(1);
        expect(onSubmit).toBeCalledWith({ q: 'cats' }, expect.anything());
    });

});

