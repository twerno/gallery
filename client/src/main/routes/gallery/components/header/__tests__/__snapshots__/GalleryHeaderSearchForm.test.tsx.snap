// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryHeaderSearchForm [GalleryHeaderSearchForm] Component basic tests matches the snapshot 1`] = `
<styled.form
  onSubmit={[Function]}
>
  <styled.input
    aria-label="Search"
    id="q"
    name="q"
    onChange={[Function]}
    type="text"
    value="cats"
  />
  <styled.button
    disabled={false}
    mode="inputButton"
    type="submit"
  >
    Search
  </styled.button>
</styled.form>
`;

exports[`GalleryHeaderSearchForm [GalleryHeaderSearchForm] Component basic tests matches the snapshot 2`] = `
<styled.form
  onSubmit={[Function]}
>
  <styled.input
    aria-label="Search"
    id="q"
    name="q"
    onChange={[Function]}
    type="text"
    value=""
  />
  <styled.button
    disabled={false}
    mode="inputButton"
    type="submit"
  >
    Search
  </styled.button>
</styled.form>
`;
